Software/vortex_ws/src/motion_planning/README.md

```diff
@@ -0,0 +1,215 @@
+
+* The Local Planner  :generate a set of control commands that‏ moves the AUV on the path generated by the Global Planner but with taking the dynamics into consideration while also trying to avoid obstacles.
+
+* motion control :translate the set of control commands from the Local Planner into actions for the thrusters‏
+
+### License
+
+The source code is released under a [GNU GENERAL PUBLIC LICENSE].
+
+**Author: vortex-co<br />
+Affiliation: [VorteX-Co](https://vortex-co.com/home)<br />
+Maintainer: vortex-co, info@vortex-co.com**
+
+The Vision module package has been tested under [ROS2] eloquent on Ubuntu 18.04 
+
+## Table of contents
+* [Installation](#Installation)
+* [Usage](#Usage)
+* [Config files](#Config-files)
+* [Launch files](#Launch-files)
+*  [Nodes](#Nodes)
+* [Hardware](#hardware)
+
+
+## Installation
+### Installation from Packages
+
+Open a terminal, clone the repository
+
+    cd ~/ros2_ws/ #use your current ros2 workspace folder
+    git clone https://github.com/VorteX-co/VAUV.git
+    
+
+### Building from Source
+
+#### Dependencies
+
+- [Robot Operating System (ROS)](https://index.ros.org/doc/ros2/) (middleware for robotics),
+-  colcon	
+	
+        sudo apt install python3-colcon-common-extensions
+
+#### Building
+
+To build from source, clone the latest version from this repository into your  workspace and compile the package using
+
+       
+     cd ~/ros2_ws/VAUV/Software/vortex_ws/src/motion_planning/src/
+     source /opt/ros/eloquent/setup.bash
+     colcon build --symlink-install
+
+
+
+## Usage
+ First go to project location and source it
+
+    cd ~/ros2_ws/VAUV/Software/vortex_ws/src/motion_planning/src/
+    source install/setup.bash
+
+   * Then we can start node by node with
+
+	ros2 run motion_planning class_name
+
+   * or launch file with
+
+	ros2 launch motion_planning launch_file_name.launch.
+
+
+## Config files
+
+
+
+
+## Launch files
+
+
+ 
+     
+
+## Nodes
+
+
+### pose_estimation node 
+estimates 3D state of the AUV with the usage of DVl,IMU,point cloud data and visual odometry from the streo camera .
+
+
+#### Subscribed Topics
+* fused_date
+
+#### Published Topics
+* estimated_pose
+
+#### Services
+..
+
+
+#### Parameters
+....
+
+### local_SLAM node    
+generates a series of poses and submaps by scan matching and using a motion filter .
+#### Subscribed Topics
+* estimated_pose‏
+
+#### Published Topics
+* submaps
+#### Services
+
+...
+
+#### Parameters‏
+...
+
+
+### global_SlAM node  
+updates the poses/states from the Local SLAM system and then updates the submaps and create a Global 3D map .
+
+#### Subscribed Topics
+* submaps
+#### Published Topics
+* Global_3D_map
+
+#### Services
+...
+
+
+#### Parameters
+...
+
+### searching_algorthim node
+it is an algorithm used for estimating the best path from starting point to goal point and also validates the generated path(validation rate ).
+
```

Comment:
The part of "it is an algorithm" should be removed and what is validation rate ?
For example
It aims to find the optimum path from the AUV current location to it is goal and continuously validating this path